cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -0,0 +1,117 @@
+# Various Ruby helper methods to generate Varnish VCL from the base configuration.
+
+# Basic regex matcher for the optional query part of a URL.
+QUERY_REGEX = "(\\?.*)?$"
+
+def path_to_regex(path)
+  path = "^/#{path.sub(/^\//,'')}"
+  if path[-1] == '*'
+    path.slice!(-1)
+  else
+    path = path + QUERY_REGEX
+  end
+  path
+end
+
+# Varnish uses 'bereq' in the 'response' section, 'req' otherwise.
+def req(section)
+  section == 'response' ? 'bereq' : 'req'
+end
+
+# Returns a regex-matcher string based on an array of filename extensions.
+def extensions_to_regex(exts)
+  return [] if exts.empty?
+  ["(?i)(#{exts.join('|').gsub('.','\.')})#{QUERY_REGEX}"]
+end
+
+# Returns a regex-conditional string fragment based on the provided behavior.
+# In the 'proxy' section, ignore extension-based behaviors (e.g., *.png).
+def paths_to_regex(path_config, section)
+  path_config = [path_config] unless path_config.is_a?(Array)
+  extensions, paths = path_config.partition{ |path| path[0] == '*' }
```

Comment:
Consider also the invalid path  */images which will be considered as an extension by this logic.